stdlib/FileWatching/src/FileWatching.jl

```diff
@@ -232,6 +235,10 @@ mutable struct _FDWatcher
     @static if Sys.iswindows()
         _FDWatcher(fd::RawFD, mask::FDEvent) = _FDWatcher(fd, mask.readable, mask.writable)
         function _FDWatcher(fd::RawFD, readable::Bool, writable::Bool)
+            if fd == RawFD(-1)
+                throw(ArgumentError("Passed file descriptor is $(fd) == RawFD(-1), this is probably not a valid file descriptor"))
+            end
+
             handle = Libc._get_osfhandle(fd)
             return _FDWatcher(handle, readable, writable)
         end
```

Comment:
And github won't let me comment on the following line, but adding the same to the next function too:
```julia
        function _FDWatcher(fd::WindowsRawSocket, readable::Bool, writable::Bool)
            if fd == Base.INVALID_OS_HANDLE
                throw(ArgumentError("file descriptor fd=$fd is not a valid file descriptor"))
            elseif !readable && !writable
```

Reply:
Added in ed06cc419c.